Update serial.markdown

Tom-Bom-badil opened the pull request:
Added note for socat & ser2net to prevent future users searching for hours to get it up and running.

<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).

  Before submitting your pull request, please verify that you have chosen the correct target branch,
  and that the PR preview looks fine and does not include unrelated changes.
-->
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->



## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [ ] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **Documentation**
	- Added a side note clarifying the availability of `ser2net` and `socat` on HASS OS for the `serial` sensor integration.
	- Ensured comprehensive instructions remain intact for setting up and utilizing the `serial` sensor integration.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Tom-Bom-badil committed:
Update serial.markdown

Added note for socat & ser2net to prevent future users searching for hours to get it up and running.

frenck reviewed (approved):
The CI is failing, could you please take a look?

Thanks! 👍 

../Frenck

frenck reviewed (changes requested):
Wrong button, trying again.

Tom-Bom-badil committed:
Update serial.markdown

Added 'not', as my initial change was not specific enough.

I am using socat since more than 10 years on other systems. 

Reason of the PR is that the reference to socat and ser2net in this doc already confused a lot of people and made them waste their time (including myself - 2 long evenings of research and try-and-error, without any result).

My conclusion is the same as a bunch of other people had before: There is not legit / stable way to integrate it into HASS OS, as it requires additional driver installations, which are lost on next reboot.
